homeassistant/components/srp_energy/config_flow.py

```diff
@@ -1,66 +1,93 @@
-        vol.Required(CONF_USERNAME): str,
-        vol.Required(CONF_PASSWORD): str,
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): str,
-        vol.Optional(CONF_IS_TOU, default=False): bool,
-    }
+    is_valid = await hass.async_add_executor_job(srp_client.validate)
 
-    async def async_step_user(self, user_input=None):
-        """Handle a flow initialized by the user."""
-        errors = {}
+    LOGGER.debug("Is valid User Input with client: %s", is_valid)
+    if not is_valid:
+        raise InvalidAuth
 
-        if self._async_current_entries():
-            return self.async_abort(reason="single_instance_allowed")
+    return is_valid
 
-        if user_input is not None:
-            try:
 
-                srp_client = SrpEnergyClient(
-                    user_input[CONF_ID],
-                    user_input[CONF_USERNAME],
-                    user_input[CONF_PASSWORD],
-                )
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle an SRP Energy config flow."""
 
-                is_valid = await self.hass.async_add_executor_job(srp_client.validate)
+    VERSION = 1
 
-                if is_valid:
-                    return self.async_create_entry(
-                        title=user_input[CONF_NAME], data=user_input
-                    )
+    @callback
+    def _show_form(self, errors: dict[str, Any] | None = None) -> FlowResult:
+        """Show the form to the user."""
+        LOGGER.debug("Show Form")
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_ID): str,
+                    vol.Required(CONF_USERNAME): str,
+                    vol.Required(CONF_PASSWORD): str,
+                    vol.Optional(
+                        CONF_NAME, default=self.hass.config.location_name
+                    ): str,
+                    vol.Optional(CONF_IS_TOU, default=False): bool,
+                }
+            ),
+            errors=errors or {},
+        )
 
-                errors["base"] = "invalid_auth"
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+        LOGGER.debug("Config entry")
+        errors = {}
 
+        if user_input:
+            try:
+                await validate_input(self.hass, user_input)
             except ValueError:
                 errors["base"] = "invalid_account"
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
             except Exception:  # pylint: disable=broad-except
-                _LOGGER.exception("Unexpected exception")
-                errors["base"] = "unknown"
+                LOGGER.exception("Unexpected exception")
+                return self.async_abort(reason="unknown")
+            else:
+                await self.async_set_unique_id(user_input.get(CONF_ID))
```

Comment:
The id is required. Don't use `dict.get` for required keys. Use `dict[key]`.

Reply:
Will fix